embrio-async/src/lib.rs

```diff
@@ -81,10 +92,8 @@ where
         // using the pointer so it is safe to dereference).
         let this = unsafe { Pin::get_unchecked_mut(self) };
         if let FutureImplState::Started(g) = &mut this.state {
-            // https://github.com/rust-lang/rust-clippy/issues/2906
-            #[allow(clippy::useless_transmute)]
             unsafe {
-                this.context = mem::transmute(cx);
+                ptr::write(&mut this.context, loosen_context_lifetime(cx));
```

Comment:
My reason for using `ptr::write` here is to avoid accessing `this.context` outside of raw pointer manipulation. I don't think it matters for soundness, but since there is a `*mut` to this variable currently live in the `UnsafeContextRef` I feel like it's cleaner to only ever access it via a raw pointer (really to do that I should be doing `*(&mut this.context as *mut _) = loosen_context_lifetime(cx);`, but `ptr::write` is a little easier to read than that, and as you mention is equivalent because of lack of drop).

Let me know if that makes any sense, or if not I'll revert this change.

Reply:
IMO, trying to only access `context` through raw ptr manipulation just makes it harder to understand. `ptr::write` is definitely better than `*(&mut this.context as *mut _)` though.

Up to you whether to keep it or not, but a comment explaining why it's being done that way will help prevent some wtf's from future readers.